modules/analysis/aws-batch-demo/stack.py

```diff
@@ -0,0 +1,175 @@
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import logging
+from typing import Any, Dict, cast
+
+import aws_cdk.aws_ec2 as ec2
+import aws_cdk.aws_ecr as ecr
+import aws_cdk.aws_iam as iam
+import cdk_nag
+from aws_cdk import Aspects, Duration, RemovalPolicy, Stack, Tags
+from aws_cdk import aws_dynamodb as dynamo
+from cdk_nag import NagSuppressions
+from constructs import Construct, IConstruct
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+class AwsBatchPipeline(Stack):  # type: ignore
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        *,
+        config: Dict[str, Any],
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(
+            scope,
+            id,
+            **kwargs,
+        )
+
+        for k, v in config.items():
+            setattr(self, k, v)
+
+        Tags.of(scope=cast(IConstruct, self)).add(
+            key="Deployment",
+            value="aws",
+        )
+
+        dep_mod = f"addf-{self.deployment_name}-{self.module_name}"
+
+        self.vpc = ec2.Vpc.from_lookup(
+            self,
+            "VPC",
+            vpc_id=self.vpc_id,
+        )
+
+        self.repository_name = dep_mod
+        ecr.Repository(self, id=self.repository_name, repository_name=self.repository_name)
+
+        # DYNAMODB TRACKING TABLE
+        self.tracking_table_name = f"{dep_mod}-drive-tracking"
+        tracking_partition_key = "pk"  # batch_id or drive_id
+        tracking_sort_key = "sk"  # batch_id / array_index_id   or drive_id / file_part
+
+        dynamo.Table(
+            self,
+            self.tracking_table_name,
+            table_name=self.tracking_table_name,
+            partition_key=dynamo.Attribute(name=tracking_partition_key, type=dynamo.AttributeType.STRING),
+            sort_key=dynamo.Attribute(name=tracking_sort_key, type=dynamo.AttributeType.STRING),
+            billing_mode=dynamo.BillingMode.PAY_PER_REQUEST,
+            removal_policy=RemovalPolicy.DESTROY,
+            point_in_time_recovery=True,
+            stream=dynamo.StreamViewType.NEW_AND_OLD_IMAGES,
+        )
+
+        # Create Dag IAM Role and policy
+        policy_statements = [
+            iam.PolicyStatement(
+                actions=["dynamodb:*"],
+                effect=iam.Effect.ALLOW,
+                resources=[f"arn:aws:dynamodb:{self.region}:{self.account}:table/{dep_mod}*"],
+            ),
+            iam.PolicyStatement(
+                actions=["ecr:*"],
+                effect=iam.Effect.ALLOW,
+                resources=[f"arn:aws:ecr:{self.region}:{self.account}:repository/{dep_mod}*"],
+            ),
+            iam.PolicyStatement(
+                actions=["s3:*"],
```

Comment:
This is giving the role (and whom assumes it) full control over all actions in all S3 buckets...is this intended?

Reply:
Removed - s3 permissions covered by the datalake-buckets full_access_policy